### PR TITLE
Migrate to pnpm

### PR DESCRIPTION
**Pull request by int128**
Align to the upstream of https://github.com/int128/typescript-actions-monorepo/pull/785.

## Changes
- Migrate the package manager to pnpm.
- Migrate Yarn workspace to pnpm workspace. See https://pnpm.io/workspaces
- Add the missing dependencies, because pnpm handles the dependencies strictly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -105,21 +105,15 @@ destination-repository  (branch: ns/${source-repository}/${overlay}/${namespace}
         └── generated.yaml
 ```
 
-## Actions
```
Comment (int128) [inline README.md:108]: Remove the outdated section.

File: .github/workflows/autofix.yaml [hunk 2]
```diff
@@ -5,6 +5,7 @@ on:
     paths:
       - '**/*.ts'
       - '*.json'
+      - '*.lock'
```
Comment (int128) [inline .github/workflows/autofix.yaml:8]: Run the workflow when the lockfile is changed.
(Same change for all workflows)

File: .github/workflows/autofix.yaml [hunk 3]
```diff
@@ -16,8 +17,8 @@ jobs:
...
         with:
           node-version: 20
-          cache: yarn
-      - run: yarn
-      - run: yarn format
-      - run: yarn lint
+      - run: corepack enable pnpm
+      - run: pnpm i
+      - run: pnpm format
+      - run: pnpm lint
```
Comment (int128) [inline .github/workflows/autofix.yaml:23]: Replace `yarn` with `pnpm`.
(Same change for all workflows)

File: bootstrap-pull-request/package.json [hunk 4]
```diff
@@ -2,14 +2,18 @@
...
-    "@actions/core": "1.10.1"
+    "@actions/core": "1.10.1",
+    "@actions/exec": "1.1.1",
+    "@actions/github": "6.0.0",
+    "@actions/io": "1.1.3",
+    "js-yaml": "4.1.0"
+  },
+  "devDependencies": {
+    "@types/js-yaml": "4.0.9"
   }
```
Comment (int128) [inline bootstrap-pull-request/package.json:18]: Add the missing dependencies.

File: bootstrap-pull-request/package.json [hunk 5]
```diff
@@ -2,14 +2,18 @@
   "name": "bootstrap-pull-request",
   "version": "0.0.0",
   "private": true,
-  "main": "lib/src/main.js",
   "scripts": {
-    "build": "tsc",
-    "package": "ncc build --source-map --license licenses.txt",
-    "test": "jest",
-    "ci:package": "yarn build && yarn package"
+    "build": "ncc build --source-map --license licenses.txt src/main.ts",
```
Comment (int128) [inline bootstrap-pull-request/package.json:6]: Change to build typescript by ncc without tsc.

File: package.json [hunk 6]
```diff
@@ -20,9 +20,6 @@
     "format": "prettier --write **/*.ts",
     "lint": "eslint --fix **/*.ts"
   },
-  "workspaces": [
-    "*"
-  ],
```
Comment (int128) [inline package.json:25]: Remove Yarn workspaces.

File: .github/workflows/autofix.yaml [hunk 7]
```diff
@@ -5,6 +5,7 @@ on:
     paths:
       - '**/*.ts'
       - '*.json'
+      - '*.yaml'
```
Comment (int128) [inline .github/workflows/autofix.yaml:8]: To trigger the workflow for `pnpm-lock.yaml` or `pnpm-workspace.yaml`.

**Review by aoi1 — APPROVED** [review]

**Review by panicboat — APPROVED** [review]

**Pull request merged**